modules/internal/ChapelArray.chpl

```diff
@@ -2219,8 +2235,11 @@ module ChapelArray {
       chpl__assertSingleArrayDomain("remove count");
       const lo = this.domain.low,
             hi = this.domain.high-count;
-      if pos > hi then
-        halt("index ", pos+count, " is outside the supported range");
+      if boundsChecking && pos+count-1 > this.domain.high then
+        halt("remove at position ", pos+count-1, " out of bounds");
+      if boundsChecking && pos < lo then
+        halt("remove at position ", pos, " out of bounds");
+
```

Comment:
@daviditen, please confirm this resolution of the merge conflict (compare to #3744)- thanks!